pkg/daemon/daemon.go

```diff
@@ -1519,6 +1553,44 @@ func removeIgnitionArtifacts() error {
 	return nil
 }
 
+// MaybePersistNetworkInterfaces runs if the host is RHEL8, which can happen
+// when scaling up older bootimages and targeting 4.13+ (rhel9).  In this case,
+// we may want to pin NIC interface names that reference static IP addresses.
+// More information in https://issues.redhat.com/browse/OCPBUGS-10787
+func MaybePersistNetworkInterfaces(osRoot string) error {
+	hostos, err := osrelease.GetHostRunningOSFromRoot(osRoot)
+	if err != nil {
+		return fmt.Errorf("checking operating system: %w", err)
+	}
+
+	nmstateBinary := "/usr/bin/nmstatectl"
+	// If we're already chrooted into the host / in the MCD case, then we
+	// need to find the binary in our saved copy of /usr/bin from the host.
+	if osRoot == "/" {
+		nmstateBinary = filepath.Join(originalContainerBin, "nmstatectl")
+	}
+
+	cmd := exec.Command(nmstateBinary, "persist-nic-names", "--root", osRoot)
```

Comment:
This looks like we are running this on all RHCOS 9-> 9 update too. Shouldn't we run persist-nic-names only when a host has been upgraded from RHCOS 8 to 9?

Reply:
In theory, yes.  In practice, we really want to be able to immediately inspect the prior state.  

So what the logic here is doing is using the `--inspect` argument which just prints out the state it wrote.  Without doing this, the logs from the pin process will be somewhere in the distant past in the journal, and possibly rotated out quickly.

With this, we can just see in the MCD logs always what happened particularly across an upgrade.

Like a lot of logging, this will be really useful *now* and then 2 years from now, probably less so...

Reply:
thanks, makes sense. I believe this is not a cpu/memory intensive operation, so should be fine.